code/model1.py

```diff
@@ -77,8 +86,33 @@ def train(self, emails, w2v):
                     # change weights based on gradient value
                     optimizer.step()
                     epoch_loss += loss.data.numpy()
+            end = time.time()
+            print 'time taken ', (end-start)
             print 'loss in epoch ' + str(epoch) + ' = ' + str(epoch_loss)
 
+        print 'Number of entries in the dictionary ', len(self.emailid_train_freq)
+        email_ids, embs = self.extract_user_embeddings()
+        utils.plot_with_tsne(email_ids, embs, True)
+
+    def extract_user_embeddings(self, threshold=1):
```

Comment:
@rohinikapoor @sarankrishna920 
If we code model2 in a way such that this works for model2 as well, 
then we can consider pulling these out into one common place instead of it being replicated across the 3 models